src/Tasks/Hash.cs

```diff
@@ -37,28 +38,42 @@ public override bool Execute()
         {
             if (ItemsToHash != null && ItemsToHash.Length > 0)
             {
-                StringBuilder hashInput = new StringBuilder();
-
-                foreach (var item in ItemsToHash)
-                {
-                    hashInput.Append(item.ItemSpec);
-                    hashInput.Append(ItemSeparatorCharacter);
-                }
-
                 using (var sha1 = SHA1.Create())
```

Comment:
Did you move the SHA using up for a reason? It looks like you could put it inside the stringbuilder borrow.